modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/extension/RepositoryLookupExtensionModelGenerator.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.artifact.activation.internal.extension;
+
+import org.mule.runtime.api.deployment.meta.MulePluginModel;
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.api.exception.MuleRuntimeException;
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.extension.api.loader.ExtensionModelLoader;
+import org.mule.runtime.module.artifact.activation.api.extension.ExtensionDiscoveryRequest;
+import org.mule.runtime.module.artifact.activation.api.extension.ExtensionModelGenerator;
+import org.mule.runtime.module.artifact.activation.api.extension.ExtensionModelLoaderRepository;
+import org.mule.runtime.module.artifact.api.classloader.ArtifactClassLoader;
+import org.mule.runtime.module.artifact.api.descriptor.ArtifactPluginDescriptor;
+import org.mule.runtime.module.artifact.api.plugin.LoaderDescriber;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static java.lang.String.format;
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.singletonMap;
+import static org.mule.runtime.api.dsl.DslResolvingContext.getDefault;
+import static org.mule.runtime.core.api.lifecycle.LifecycleUtils.startIfNeeded;
+import static org.mule.runtime.extension.api.loader.ExtensionModelLoadingRequest.builder;
+
+/**
+ * Generates an extension model by delegating to the appropriate {@link ExtensionModelLoader}.
+ *
+ * @since 4.5
+ */
+public class RepositoryLookupExtensionModelGenerator implements ExtensionModelGenerator {
```

Comment:
I'm having trouble making sense of the introduction of this class. This logic is currently part of `org.mule.runtime.deployment.model.api.artifact.extension.ExtensionModelDiscoverer`. What's the point of separating it here? You also introduced an activation class called ExtensionModelDiscoverer, and what I expect from a class with that name is to yield extension models. I don't see the need for this class. Can you please explain?